pandas/core/indexes/base.py

```diff
@@ -2319,7 +2319,7 @@ def union(self, other, sort=None):
         else:
             rvals = other._values
 
-        if self.is_monotonic and other.is_monotonic:
+        if self.is_monotonic and other.is_monotonic and sort is None:
```

Comment:
I will need to check more carefully whether adding this affects any of the other index types because at the moment it only causes a small number of ``PeriodIndex`` and ``DatetimeIndex`` tests to fail.

Reply:
can you move the sort check to the beginning